src/guess_density.jl

```diff
@@ -93,23 +94,23 @@ which follow the functional form
 and are placed at `position` (in fractional coordinates).
 """
 function gaussian_superposition(basis::PlaneWaveBasis{T}, gaussians) where {T}
-    ρ = zeros(complex(T), basis.fft_size)
-    isempty(gaussians) && return G_to_r(basis, ρ)
-
-    # Fill ρ with the (unnormalized) Fourier transform, i.e. ∫ e^{-iGx} f(x) dx,
-    # where f(x) is a weighted gaussian
-    #
-    # is formed from a superposition of atomic densities, each scaled by a prefactor
-    for (iG, G) in enumerate(G_vectors(basis))
-        Gsq = sum(abs2, basis.model.recip_lattice * G)
+    ρ = deepcopy(basis.G_vectors)
+    #These copies are required so that recip_lattice and gaussians are isbits (GPU compatibility)
+    recip_lattice = basis.model.recip_lattice
```

Comment:
This is not a copy?

Reply:
You might be running into the big bad "captured variable in closure" performance bug

Reply:
Actually, I realised we don't need that copy at all. Later on, I am forced to use `map` instead of `map!` (we can't use the inplace version as we are converting an Array of Vec3 to an Array of Complex), so a new array is build there. So I can just remove this line and put  `ρ = map(build_ρ, basis.G_vectors)` instead of  `ρ = map(build_ρ, ρ)`.